More-like-this solr queries should filter out unpublished works.

eddierubeiz opened the pull request:
Ref #2998

jrochkind commented:
Love it. Surprised the value is `1`, not `true`.  Do we need/want a test that this is working right, and is succesfully filtering things out?

eddierubeiz commented:
I changed it to true, although 1 also does work (see e.g. [this other code](https://github.com/sciencehistory/scihist_digicoll/blob/master/app/models/search_builder/access_control_filter.rb)). I also added an explicit test for that part of the code -- good thinking.
